include/kernel/thread.h

```diff
@@ -145,7 +145,7 @@ static inline void commitTime(void)
         }
         NODE_STATE(ksCurSC)->scConsumed += NODE_STATE(ksConsumed);
     }
-    if (CONFIG_NUM_DOMAINS > 1) {
+    if (CONFIG_NUM_DOMAINS > 1 && !switched_domain) {
```

Comment:
Nice. I think this achieves what we need. Could you please add a comment that explains why we need `switched_domain`?

Maybe
``/* If we have switched domains, ksDomainTime has already been reset and is accounted for the old domain. If we have not switched domains, we still need to account for the time that has passed. */